slack_command_bot/component.py

```diff
@@ -105,27 +106,26 @@ def init_flask_app(self):
         )
 
         self._create_oauth_url(
-            flask_app=self._flask_app,
+            flask_app=app,
             slack_client_id=self.slack_client_id,
             state_store=state_store,
             authorize_url_generator=authorize_url_generator,
             installation_store=installation_store,
         )
         self._create_redirect_url(
-            flask_app=self._flask_app,
+            flask_app=app,
             slack_client_id=self.slack_client_id,
             client_secret=self.client_secret,
             state_store=state_store,
             installation_store=installation_store,
         )
+        app.route(self.command, methods=["POST", "GET"])(self.handle_command)
 
     def run(self, *args, **kwargs) -> None:
-        self.init_flask_app()
-        self._flask_app.route(self.command, methods=["POST", "GET"])(
-            self.handle_command
-        )
+        app = Flask(__name__)
```

Comment:
if `app` is sure to be created before `init_flask_app(...)` then why do we make it an optional argument?

Reply:
it's this
```
    def init_flask_app(self, app: Flask = None):
        if not app:
            app = Flask(__name__)
```